internal/rpc/rpcserver/server.go

```diff
@@ -3469,3 +3470,48 @@ func (s *walletServer) LockAccount(ctx context.Context, req *pb.LockAccountReque
 	}
 	return &pb.LockAccountResponse{}, nil
 }
+
+// getPeerInfo responds to the getpeerinfo request.
+// It gets the network backend and views the data on remote peers when in spv mode
+func (s *walletServer) GetPeerInfo(ctx context.Context, req *pb.GetPeerInfoRequest) (*pb.GetPeerInfoResponse, error) {
+	n, err := s.wallet.NetworkBackend()
+	if err != nil {
+		return nil, err
+	}
+	syncer, ok := n.(*spv.Syncer)
+	if !ok {
+		var resp []*pb.GetPeerInfoResponse_PeerInfo
+		if rpc, ok := n.(*dcrd.RPC); ok {
+			err := rpc.Call(ctx, "getpeerinfo", &resp)
```

Comment:
This is wrong, you are passing the (autogenerated) struct variable describing a protobuf message to the json-rpc client, and dcrd may not return a result that unmarshals correctly into these fields.